### PR TITLE
Missing Maven Updates

### PR DESCRIPTION
**Pull request by mitchej123**
* Add Maven and missing maven notification
* Add Repo Blacklist

### CODE REVIEW

**Comment by mitchej123** [comment]
```
Grabbing all repository information
****** Missing Mods: Infernal-Mobs, LunatriusCore, Forgelin, AngerMod, StorageDrawers-Forestry, GTSU, LittleTiles, Catwalks-2, StorageDrawers-Natura, AFSU, RemoteIO, StandardQuestingPack, GTNEIOrePlugin, NEI-Integration, ForgeGradle, ToroHealth, ModTweaker, MagicBees, OpenCC-3D-printer-models, Tinkers-Defense, SpecialMobs, Universal-Singularities, ThaumicHorizons, NotEnoughThaumcraftTabs, Minecraft-Backpack-Mod, StorageDrawers-BiomesOPlenty, FindIt, Tainted-Magic, Railcraft-API, ThaumcraftMobAspects, DefaultWorldGenerator, CompactKineticGenerators, OpenModsLib, Schematica, ModMixins, CreativeCore, Roguelike-Dungeons, Sanguimancy, TooMuchLoot, CraftGuide, ForbiddenMagic, Super-TiC, HydroEnergy, BloodyQuests, EnhancedLootBags, GTNH-Lanthanides, Enklume, OverloadedArmorBar, CTMLib, MoarSigns, TiC-Tooltips, Random-Things, thaumcraft-research-tweaks, LogisticsPipes, Nodal-Mechanics, NotEnoughProduction, SGCraft, NetherPortalFix, IFU, SpongeMixins, StorageDrawers-Misc, DefaultServerList, supersolarpanels, OCGlasses, SpecialAI, BlockLimiter, Minetweaker-Gregtech-5-Addon, oauth
****** Missing Maven: WitcheryExtras, GTNH-TC-Wands, Hardcore-Ender-Expansion, Binnie, EnderZoo, Irontanks, LootGames, BetterQuesting, ae2stuff, DetravScannerMod, ThaumicBases, TC4Tweaks, WAWLA, SpiceOfLife, VisualProspecting, KekzTech, Gadomancy, OpenBlocks, WailaHarvestability, Botanic-horizons, nei-custom-diagram, TCNodeTracker, Electro-Magic-Tools, ArchitectureCraft, StevesAddons, Hodgepodge, WitchingGadgets, BetterLoadingScreen, InGame-Info-XML, TCNEIAdditions, neiaddons, Netty-Patcher, Steve-s-Factory-Manager
```

File: repo-blacklist.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  "SpongePoweredMixin",
+  "QuestBook",
+  "GTNewHorizons.github.io",
+  "NewHorizonsServerCore",
+  "The-Aether-GTNH",
+  "MyEssentials-Core",
+  "NewHorizonsTweaker",
+  "SpongePoweredMixinUpstream",
+  "ASMHelper",
+  "BugTorch",
```
Comment (boubou19) [inline repo-blacklist.json:25]: it's in the pack

File: repo-blacklist.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  "TravellersGear-Private",
+  "BON2",
+  "ImpactArchitectureModule",
+  "ForgePatcher",
+  "BQ3-Thermos-patch",
+  "Botdustries",
+  "TickProfiler",
+  "GTNH-O-Plenty",
+  "SpongePoweredMixin",
+  "QuestBook",
```
Comment (boubou19) [inline repo-blacklist.json:17]: isn't it the mod adding the questbook item? if yes it's in the pack

File: repo-blacklist.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  "TinkersGregworks",
+  "DreamAssemblerXXL",
+  "Alfheim",
+  "MicdoodleCore",
+  "AdvancedSystemsManager",
+  "HungerOverhaul",
+  "MyTown2",
+  "thaumcraft-api",
+  "MyPermissions",
+  "Opis",
```
Comment (bombcar) [inline repo-blacklist.json:54]: Opis is server-side and harmless on client side imo.

Comment (boubou19) [inline repo-blacklist.json:54]: we should put this in the pack as it's one of the best tools to spot lag. SP players will have some in game way to audit their map

File: repo-blacklist.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  "D3Log",
+  "TravellersGear-Private",
+  "BON2",
+  "ImpactArchitectureModule",
+  "ForgePatcher",
+  "BQ3-Thermos-patch",
+  "Botdustries",
+  "TickProfiler",
+  "GTNH-O-Plenty",
+  "SpongePoweredMixin",
```
Comment (bombcar) [inline repo-blacklist.json:16]: example of an important dependency for some mods, but not a mod itself

File: repo-blacklist.json (hunk 1 above)
Comment (bombcar) [inline repo-blacklist.json:25]: we use this iirc

**Review by bombcar — APPROVED** [review]

File: src/gtnh/utils.py [hunk 5]
```diff
@@ -91,3 +97,24 @@ def get_mod_asset(release: GitRelease) -> GitReleaseAsset:
             continue
 
         return asset
+
+
+def get_maven(mod_name: str) -> Optional[str]:
+    maven_url = MAVEN_BASE_URL + mod_name + "/"
+    response = requests.head(maven_url)
+
+    return maven_url if response.status_code == 200 else None
```
Comment (boubou19) [inline src/gtnh/utils.py:106]: could be cool to log a warn if the status code is a 5xx to let the user know it's not missing but unreachable

File: repo-blacklist.json (hunk 1 above)
Comment (mitchej123) [reply repo-blacklist.json:25]: whoops meant to remove that

**Pull request merged**